### PR TITLE
Unqualified lookup fixes, part 2

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <rdar://problem/22587551>, <rdar://problem/44777661>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@nkcsgexi Please take a look at the swift-api-digester changes.

File: tools/swift-api-digester/ModuleAnalyzerNodes.cpp [hunk 1]
```diff
@@ -658,9 +658,18 @@ bool static isSDKNodeEqual(SDKContext &Ctx, const SDKNode &L, const SDKNode &R)
...
   auto *RightAlias = dyn_cast<SDKNodeTypeAlias>(&R);
   if (LeftAlias || RightAlias) {
+    auto Left = L.getAs<SDKNodeType>();
+    auto Right = R.getAs<SDKNodeType>();
+
+    // First compare the parameter attributes.
+    if (Left->hasDefaultArgument() != Right->hasDefaultArgument())
+      return false;
+    if (Left->getParamValueOwnership() != Right->getParamValueOwnership())
+      return false;
```
[the remark this answers is not in the document]
Comment (slavapestov) [reply tools/swift-api-digester/ModuleAnalyzerNodes.cpp]: Can I do it in a follow-up patch?

[the remark this answers is not in the document]
Comment (slavapestov) [reply tools/swift-api-digester/ModuleAnalyzerNodes.cpp]: Done

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: tools/swift-api-digester/ModuleAnalyzerNodes.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (nkcsgexi) [reply tools/swift-api-digester/ModuleAnalyzerNodes.cpp]: Thanks you!

**Comment by slavapestov** [comment]
```
22:07:36 ld: warning: Could not find auto-linked library 'swiftMetal'
22:07:36 Undefined symbols for architecture x86_64:
22:07:36   "__swift_FORCE_LOAD_$_swiftMetal", referenced from:
22:07:36       __swift_FORCE_LOAD_$_swiftMetal_$_Vision in Vision.o
22:07:36      (maybe you meant: __swift_FORCE_LOAD_$_swiftMetal_$_Vision)
22:07:36 ld: symbol(s) not found for architecture x86_64
```

@shahmishal This problem is back... any ideas?

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Pull request merged**